canopy_main.ml

```diff
@@ -38,9 +38,34 @@ module Main  (C: CONSOLE) (RES: Resolver_lwt.S) (CON: Conduit_mirage.S) (S:Cohtt
 	 let body = List.fold_left (fun a b -> a ^ "\n" ^ b) "" errors in
 	 S.respond_string ~status:`Bad_request ~body () in
 
+    let update_atom, get_atom =
+      let cache = ref None in
+      let update_atom () =
+        let l = KeyHashtbl.fold (fun key x acc -> x :: acc) content_hashtbl []
+                |> List.sort Canopy_content.compare
+                |> Canopy_utils.resize 10 in
+        let entries = List.map Canopy_content.to_atom l in
+        let ns_prefix _ = Some "" in
+        Store.last_commit_date ()
+        >|= fun updated ->
+          Syndic.Atom.feed
+            ~id:(Uri.of_string config.blog_name)
```

Comment:
should a feed contain its own url? (I'm not reading feeds, and haven't read the atom spec for quite some time, but I thought this was the case).. this information is afaics not yet available inside of canopy.. maybe it should via a config parameter?